plutus-contract/src/Language/Plutus/Contract/Wallet.hs

```diff
@@ -53,7 +53,7 @@ balanceWallet utx = do
 computeBalance :: WAPI.MonadWallet m => Tx -> m Value
 computeBalance tx = (P.-) <$> left <*> pure right  where
     right = Ada.toValue (L.txFee tx) P.+ foldMap (view Tx.outValue) (tx ^. Tx.outputs)
-    left = foldM go P.zero (tx ^. Tx.inputs)
+    left = (P.+) (L.txForge tx) <$> foldM go P.zero (tx ^. Tx.inputs)
```

Comment:
```suggestion
    left = foldM go P.zero (L.txForge tx : (tx ^. Tx.inputs))
```

Reply:
wait that doesn't actually work, `Tx.inputs` is a set of `TxInRef`, and `go` looks up the value of each reference, and we have to add `txForge tx` _after_ all that. I will give `go` a better name though!